pkg/controller/atlasproject/teams.go

```diff
@@ -80,54 +80,62 @@ func (r *AtlasProjectReconciler) syncAssignedTeams(ctx *workflow.Context, projec
 
 	defer statushandler.Update(ctx, r.Client, r.EventRecorder, project)
 
-	toDelete := make([]*mongodbatlas.Result, 0, len(atlasAssignedTeams.Results))
-	for _, atlasAssignedTeam := range atlasAssignedTeams.Results {
-		desiredTeam, ok := teamsToAssign[atlasAssignedTeam.TeamID]
-		if !ok {
-			toDelete = append(toDelete, atlasAssignedTeam)
+	toDelete := make([]*admin.TeamRole, 0)
 
-			continue
-		}
+	if atlasAssignedTeams != nil && atlasAssignedTeams.Results != nil {
+		for _, atlasAssignedTeam := range *(atlasAssignedTeams.Results) {
+			var desiredTeam *akov2.Team
+			var ok bool
+			if atlasAssignedTeam.TeamId != nil {
+				desiredTeam, ok = teamsToAssign[atlasAssignedTeam.GetTeamId()]
+				if !ok {
+					result := atlasAssignedTeam
+					toDelete = append(toDelete, &result)
 
-		if !hasTeamRolesChanged(atlasAssignedTeam.RoleNames, desiredTeam.Roles) {
-			currentProjectsStatus[atlasAssignedTeam.TeamID] = status.ProjectTeamStatus{
-				ID:      atlasAssignedTeam.TeamID,
-				TeamRef: desiredTeam.TeamRef,
+					continue
+				}
```

Comment:
remove this whole block between line 87 and 96 in favor of suggestion available at https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1817/files?diff=split&w=0#r1776459297.